go-controller/pkg/cni/cni.go

```diff
@@ -199,3 +178,24 @@ func (pr *PodRequest) getCNIResult(podInterfaceInfo *PodInterfaceInfo) (*current
 		IPs:        ips,
 	}, nil
 }
+
+// getAnnotationsOnPod obtains the pod annotation from the cache
+func getAnnotationsOnPod(podLister corev1listers.PodLister, namespace, name string) (map[string]string, error) {
+	var annotations map[string]string
+
+	if err := wait.PollImmediate(1*time.Second, 5*time.Second, func() (bool, error) {
+		pod, err := podLister.Pods(namespace).Get(name)
+		if err != nil {
+			return false, err
```

Comment:
@aojea original code distinguishes between "pod not found" (hard fail) and other errors (retry). I think we should do that here too?

Reply:
there are no other errors in the cache AFAIK, but I can recover it